pkgs/os-specific/linux/gvusb2/Makefile.patch

```diff
@@ -0,0 +1,17 @@
+diff --git a/Makefile b/Makefile
+index e666392..e1a341b 100644
+--- a/Makefile
++++ b/Makefile
+@@ -3,8 +3,10 @@ obj-m += gvusb2-sound.o gvusb2-video.o
+ gvusb2-sound-y := gvusb2-snd.o gvusb2-core.o
+ gvusb2-video-y := gvusb2-vid.o gvusb2-v4l2.o gvusb2-core.o gvusb2-i2c.o
+ 
++KERNEL_SOURCE_DIR := /lib/modules/$(shell uname -r)/build
```

Comment:
Why do we need this patch? It looks like it changes nothing for me.

Reply:
The Makefile included with the project doesn't use any environment variables:
```
❯ git show 811fb0f6ee4ef62e0e346a293bb5e3b61ad10b7d:Makefile
obj-m += gvusb2-sound.o gvusb2-video.o

gvusb2-sound-y := gvusb2-snd.o gvusb2-core.o
gvusb2-video-y := gvusb2-vid.o gvusb2-v4l2.o gvusb2-core.o gvusb2-i2c.o

all:
        make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules

clean:
        make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean

```

Without this patch, the build is running the default `make -C /lib/modules/5.10.7/build M=/build/source modules` and failing:
```
❯ nix-build -A linuxPackages_latest.gvusb2
these derivations will be built:
  /nix/store/warcxpjpb5kknzkwrnbq21b50wavbz1k-gvusb2-2020-07-27-unstable.drv
building '/nix/store/warcxpjpb5kknzkwrnbq21b50wavbz1k-gvusb2-2020-07-27-unstable.drv'...
unpacking sources
unpacking source archive /nix/store/iwc4l1vs3sqxmf99cq0839dl2zh2xnn1-source
source root is source
patching sources
configuring
no configure script, doing nothing
building
build flags: SHELL=/nix/store/kl6lr3czkbnr6m5crcy8ffwfzbj8a22i-bash-4.4-p23/bin/bash KERNEL_SOURCE_DIR=/nix/store/2kxfgzzdwi1nj6axsi43r05mcd6ncqg4-linux-5.10.7-dev/lib/modules/5.10.7/build INSTALL_MOD_PATH=\$\(out\)
make -C /lib/modules/5.10.7/build M=/build/source modules
make[1]: *** /lib/modules/5.10.7/build: No such file or directory.  Stop.
make: *** [Makefile:7: all] Error 2
builder for '/nix/store/warcxpjpb5kknzkwrnbq21b50wavbz1k-gvusb2-2020-07-27-unstable.drv' failed with exit code 2
error: build of '/nix/store/warcxpjpb5kknzkwrnbq21b50wavbz1k-gvusb2-2020-07-27-unstable.drv' failed
```

I'm using https://github.com/NixOS/nixpkgs/pull/105899 as a reference along with https://nixos.wiki/wiki/Linux_kernel#Packaging_out-of-tree_kernel_modules.

I'm adding an environment variable `KERNEL_SOURCE_DIR` to the Makefile to choose the right directory during the build, using `${kernel.dev}`:
`make -C /nix/store/2kxfgzzdwi1nj6axsi43r05mcd6ncqg4-linux-5.10.7-dev/lib/modules/5.10.7/build M=/build/source modules`
```
build flags: SHELL=/nix/store/kl6lr3czkbnr6m5crcy8ffwfzbj8a22i-bash-4.4-p23/bin/bash KERNEL_SOURCE_DIR=/nix/store/2kxfgzzdwi1nj6axsi43r05mcd6ncqg4-linux-5.10.7-dev/lib/modules/5.10.7/build INSTALL_MOD_PATH=\$\(out\)
make -C /nix/store/2kxfgzzdwi1nj6axsi43r05mcd6ncqg4-linux-5.10.7-dev/lib/modules/5.10.7/build M=/build/source modules
```

I don't have much experience building kernel modules and may be missing some conventions here.